File: hashgrid/exceptions.py
"""Custom exceptions for the Hashgrid client."""


class HashgridError(Exception):
    """Base exception for all Hashgrid errors."""
    pass


class HashgridAPIError(HashgridError):
    """Exception raised for API errors."""
    
    def __init__(self, message, status_code=None, response=None):
        super().__init__(message)
        self.status_code = status_code
        self.response = response


class HashgridAuthenticationError(HashgridAPIError):
    """Exception raised for authentication errors."""
    pass


class HashgridNotFoundError(HashgridAPIError):
    """Exception raised when a resource is not found."""
    pass


class HashgridValidationError(HashgridAPIError):
    """Exception raised for validation errors."""
    pass

File: js/src/exceptions.ts
/** Custom exceptions for the Hashgrid client. */

export class HashgridError extends Error {
  constructor(message: string) {
    super(message);
    this.name = "HashgridError";
    Object.setPrototypeOf(this, HashgridError.prototype);
  }
}

export class HashgridAPIError extends HashgridError {
  statusCode?: number;
  response?: Response;

  constructor(message: string, statusCode?: number, response?: Response) {
    super(message);
    this.name = "HashgridAPIError";
    this.statusCode = statusCode;
    this.response = response;
    Object.setPrototypeOf(this, HashgridAPIError.prototype);
  }
}

export class HashgridAuthenticationError extends HashgridAPIError {
  constructor(message: string, statusCode?: number, response?: Response) {
    super(message, statusCode, response);
    this.name = "HashgridAuthenticationError";
    Object.setPrototypeOf(this, HashgridAuthenticationError.prototype);
  }
}

export class HashgridNotFoundError extends HashgridAPIError {
  constructor(message: string, statusCode?: number, response?: Response) {
    super(message, statusCode, response);
    this.name = "HashgridNotFoundError";
    Object.setPrototypeOf(this, HashgridNotFoundError.prototype);
  }
}

export class HashgridValidationError extends HashgridAPIError {
  constructor(message: string, statusCode?: number, response?: Response) {
    super(message, statusCode, response);
    this.name = "HashgridValidationError";
    Object.setPrototypeOf(this, HashgridValidationError.prototype);
  }
}
